verilog/tools/kythe/verilog_kythe_extractor.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// verilog_kythe_extractor is a simple command-line utility to extract kythe
+// indexing facts from the given file.
+//
+// Example usage:
+// verilog_kythe_extractor files...
+
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <memory>
+#include <sstream>  // IWYU pragma: keep  // for ostringstream
+#include <string>   // for string, allocator, etc
+#include <utility>
+#include <vector>
+
+#include "absl/flags/flag.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/substitute.h"
+#include "common/text/concrete_syntax_leaf.h"
+#include "common/text/concrete_syntax_tree.h"
+#include "common/text/parser_verifier.h"
+#include "common/text/symbol.h"
+#include "common/text/text_structure.h"
+#include "common/text/token_info.h"
+#include "common/util/file_util.h"
+#include "common/util/init_command_line.h"
+#include "common/util/logging.h"  // for operator<<, LOG, LogMessage, etc
+#include "verilog/CST/verilog_nonterminals.h"
+#include "verilog/CST/verilog_tree_print.h"
+#include "verilog/analysis/verilog_analyzer.h"
+#include "verilog/parser/verilog_parser.h"
+#include "verilog/tools/kythe/indexing_facts_tree_extractor.h"
+#include "verilog/tools/kythe/verilog_extractor_indexing_fact_type.h"
+
+ABSL_FLAG(bool,
+          printextraction,
+          false,
+          "Whether or not to print the extracted general indexing facts "
+          "from the middle layer)");
+
+int ExtractOneFile(absl::string_view content, absl::string_view filename) {
```

Comment:
I would even move this to a library. That will make testing much easier (you'll be able to provide a SystemVerilog file & parse it instead of manually constructing CST for tests).